epictrack-web/src/components/shared/MasterTrackTable/utils.ts

```diff
@@ -16,7 +16,14 @@ export function getSelectFilterOptions<T>(
     }
 
     // Step 4: Populate the Map with unique values and their formatted labels
-    optionsMap.set(dataObject[key], formatLabel(dataObject[key]));
+    if (["start_date", "end_date"].includes(key as string)) {
+      optionsMap.set(
+        formatLabel(dataObject[key]),
+        formatLabel(dataObject[key])
```

Comment:
I would suggest adding an optional type prop, and if type === 'date' we do this. date keys could be other than start_date or end_date

Reply:
I'm thinking even we can add a formatValue. Which could be default to () => dataOject[key]. I think this is probably a the better suggestion